src/util/address.rs

```diff
@@ -82,39 +89,32 @@ pub struct Address {
     /// The type of the address
     pub payload: Payload,
     /// The network on which this address is usable
-    pub network: Network
+    pub network: Network,
+    /// The public key corresponding to this address, if known
+    pub public_key: Option<key::PublicKey>,
```

Comment:
Hmm. If one cares about keeping it, the user can `.clone()` it, no?

Reply:
@stevenroose the goal is that the `Address` type is able to be used to get the same data that the `validateaddress` Core RPC uses. If you don't keep this data it is impossible to fill out PSBT information for addresses that you know. Requiring users clone it separately is not a reasonable requirement.

But I agree it would make more sense to move it into the `Payload` enum, specifically the `PubkeyHash` and `WitnessProgram` variants. (It's unfortunate that `WitnessProgram` does not carry this data itself, but it makes sense for bitcoin-bech32 not to depend on rust-bitcoin types.)

Reply:
Yeah it's just that an address that is generated from string will then be different then the one you create with pubkey. A created one will thus not roundtrip with `to_string`. But well fair enough.

Reply:
@apoelstra @stevenroose If we want to move `PublicKey` into the `Payload` enum, we would have to move it into the `PubkeyHash`, `ScriptHash`, and `WitnessProgram` variants. As in the case of `p2shwpkh`, the payload is a `ScriptHash`. I believe it would make more sense to keep `PublicKey` in the `Address` struct.